ast/misc.go

```diff
@@ -2196,6 +2196,207 @@ func (n *ShutdownStmt) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+type BRIEKind uint8
+type BRIEOptionType uint16
+
+const (
+	BRIEKindBackup BRIEKind = iota
+	BRIEKindRestore
+
+	// common BRIE options
+	BRIEOptionRateLimit BRIEOptionType = iota + 1
+	BRIEOptionConcurrency
+	BRIEOptionChecksum
+	BRIEOptionSendCreds
+	// backup options
+	BRIEOptionBackupTimeAgo
+	BRIEOptionLastBackupTS
+	BRIEOptionLastBackupTSO
+	// restore options
+	BRIEOptionOnline
+	// S3 storage options
+	BRIEOptionS3Endpoint
+	BRIEOptionS3Region
+	BRIEOptionS3StorageClass
+	BRIEOptionS3SSE
+	BRIEOptionS3ACL
+	BRIEOptionS3AccessKey
+	BRIEOptionS3SecretAccessKey
+	BRIEOptionS3Provider
+	BRIEOptionS3ForcePathStyle
+	BRIEOptionS3UseAccelerateEndpoint
+	// GCS storage options
+	BRIEOptionGCSEndpoint
+	BRIEOptionGCSStorageClass
+	BRIEOptionGCSPredefinedACL
+	BRIEOptionGCSCredentialsFile
+)
+
+func (kind BRIEKind) String() string {
+	switch kind {
+	case BRIEKindBackup:
+		return "BACKUP"
+	case BRIEKindRestore:
+		return "RESTORE"
+	default:
+		return ""
+	}
+}
+
+func (kind BRIEOptionType) String() string {
+	switch kind {
+	case BRIEOptionRateLimit:
+		return "RATE_LIMIT"
+	case BRIEOptionConcurrency:
+		return "CONCURRENCY"
+	case BRIEOptionChecksum:
+		return "CHECKSUM"
+	case BRIEOptionSendCreds:
+		return "SEND_CREDENTIALS_TO_TIKV"
+	case BRIEOptionBackupTimeAgo:
+		return "SNAPSHOT"
```

Comment:
should we add BRIEOptionBackupTS after [#172](https://github.com/pingcap/br/pull/172) merged

Reply:
Yes. But we also need to add a hidden config to pass the timezone.